buildSrc/src/main/kotlin/scripts/variants.gradle.kts

```diff
@@ -0,0 +1,46 @@
+package scripts
+
+plugins { id("core.android") }
+
+private enum class BuildTypes(val value: String) {
+    DEBUG("debug"), RELEASE("release")
+}
+
+private enum class ProductFlavors(val value: String) {
+    DEV("dev"), INTERNAL("internal"), PUBLIC("public")
+}
+
+private enum class FlavorDimensions(val value: String) {
+    VERSION("version")
+}
+
+android {
+    buildTypes {
+        getByName(BuildTypes.DEBUG.value) {
+            isMinifyEnabled = false
+            applicationIdSuffix = ".${BuildTypes.DEBUG.value}"
+            isDebuggable = true
+        }
+        getByName(BuildTypes.RELEASE.value) {
+            isMinifyEnabled = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+
+    flavorDimensions(FlavorDimensions.VERSION.value)
+    productFlavors {
+        create(ProductFlavors.DEV.value) {
+            dimension = FlavorDimensions.VERSION.value
+            applicationIdSuffix = ".${ProductFlavors.DEV.value}"
+            versionNameSuffix = "-${ProductFlavors.DEV.value}"
+        }
+        create(ProductFlavors.INTERNAL.value) {
+            dimension = FlavorDimensions.VERSION.value
+            applicationIdSuffix = ".${ProductFlavors.INTERNAL.value}"
+            versionNameSuffix = "-${ProductFlavors.INTERNAL.value}"
+        }
+        create(ProductFlavors.PUBLIC.value) {
```

Comment:
I think `ProductFlavors.PROD`makes more sense as a name convention for this flavor

Reply:
I considered that, but the reason why not is that for the time being we will have this for the public, but there will be other product flavors that are specific for customers. 

From my perspective: Release means Production, then you have different product flavors that satisfy different contexts. Does that make sense? How would you organize this then for custom builds for customers?

Reply:
I don't think the term **release** means **production**. In our case, we have some combination between flavors/build types like`devRelease`and `internalRelease` but don't have any relation with the production version.

Anyway, it makes sense to use term *PUBLIC* if we wanna add more flavours for our customers.

Reply:
Good point about the definition of release. And related to Public it is exactly what you pointed out. Thanks for the feedback.